libsolidity/analysis/TypeChecker.cpp

```diff
@@ -2303,11 +2303,16 @@ bool TypeChecker::visit(FunctionCall const& _functionCall)
 		functionType = dynamic_cast<FunctionType const*>(expressionType);
 		funcCallAnno.kind = FunctionCallKind::FunctionCall;
 
+		// We check for !TupleExpression as `(x.f)()` for example has
+		// `hasDeclaration() == true` due to typeforwarding for tuples with one
+		// member
+		if (functionType->hasDeclaration() && !dynamic_cast<TupleExpression const*>(&_functionCall.expression()))
```

Comment:
I think it would be best to use exactly this one: https://github.com/ethereum/solidity/pull/10384/files#diff-8e1a453b201906ae5f79d442a51974453a9a434e6631837f9b54f1c8b2781c67L807